core/dimensions.cpp

```diff
@@ -33,6 +33,11 @@ Dimensions::Dimensions(const std::vector<Dim> &labels,
     addInner(labels[i], shape[i]);
 }
 
+Dimensions::Dimensions(const Sizes &sizes) {
+  for (const auto &[label, size] : sizes)
+    addInner(label, size);
```

Comment:
Not really, but have have always done so (previously this was a constructor from `unordered_map`, this is just refactored). See also https://github.com/scipp/scipp/projects/9#card-58503236 with the thought about making `Sizes` a base class. Then it may be possible to rewrite the ops that currently use this constructor to work with `Sizes` instead.

Reply:
Maybe it should be explicit to help avoid some accidents.

Reply:
Made explicit 👍